libsolidity/codegen/ir/IRGeneratorForStatements.cpp

```diff
@@ -818,15 +818,43 @@ void IRGeneratorForStatements::endVisit(FunctionCall const& _functionCall)
 	{
 		break;
 	}
-	case FunctionType::Kind::GasLeft:
+	case FunctionType::Kind::AddMod:
+	case FunctionType::Kind::MulMod:
 	{
-		define(_functionCall) << "gas()\n";
+		static map<FunctionType::Kind, string> functions = {
+			{FunctionType::Kind::AddMod, "addmod"},
+			{FunctionType::Kind::MulMod, "mulmod"},
+		};
+		solAssert(functions.find(functionType->kind()) != functions.end(), "");
+		solAssert(arguments.size() == 3 && parameterTypes.size() == 3, "");
+
+		string var = m_context.newYulVariable();
+		m_code << "let " << var << " := " << expressionAsType(*arguments[2], *(parameterTypes[2])) << "\n";
+		Whiskers templ("if iszero(<var>) { revert(0, 0) }\n");
+		m_code << templ("var", var).render();
+
+		string args;
+		for (size_t i = 0; i < 2; ++i)
+			args += expressionAsType(*arguments[i], *(parameterTypes[i])) + ", ";
```

Comment:
Please don't do the type conversion twice! Also you could add a yul util function for these - but you don't have to.

Reply:
Ah sorry, this only converts the other two!